cmd/kube-proxy/app/server.go

```diff
@@ -150,12 +162,32 @@ func AddFlags(options *Options, fs *pflag.FlagSet) {
 	utilfeature.DefaultFeatureGate.AddFlag(fs)
 }
 
+func NewOptions() Options {
+	o := Options{
+		config:      new(componentconfig.KubeProxyConfiguration),
+		healthzPort: 10256,
+	}
+
+	var (
+		groupFactoryRegistry = make(announced.APIGroupFactoryRegistry)
+		registry             = registered.NewOrDie("")
+	)
+
+	o.scheme = runtime.NewScheme()
```

Comment:
@ncdc what about the second comment above? We are trying to remove the registry from many places, e.g. from the clients. We shouldn't rely on the static information it provides, but use discovery in kubectl/proxy/client-go instead. Here it's only about encoding, we do not even need discovery.